src/main/java/UI/ListingListPage.java

```diff
@@ -1,129 +1,100 @@
-        setBounds(length / 2, 90, length, height);
-        c.setLayout(null);
-
-        title = new JLabel("Listings");
-        title.setFont(new Font("Arial", Font.PLAIN, 30));
-        title.setSize(300, 30);
-        title.setLocation(500, 30);
-        c.add(title);
-
-
-        Listing listing = new Listing(example_user, "basketball", 40.0f,
-                "basketball for indoor use", example_images);
-
-        // Listing searchResults = new ArrayList<Listing>();
-        List <Listing> searchResults = new ArrayList<Listing>();
-
-        searchResults.add(listing);
-
-
-
-        for (Listing result : searchResults) {
-            JLabel resultTitle = new JLabel(result.getTitle());
-            JLabel resultPrice = new JLabel("Price: " + String.valueOf(result.getPrice()));
-            JLabel resultImage = new JLabel(result.getImagePath());
-            JLabel resultDescription = new JLabel(result.getDescription());
-            JLabel resultDate = new JLabel(result.getDate().toString());
-
-            resultTitle.setFont(new Font("Arial", Font.PLAIN, 10));
-            resultPrice.setFont(new Font("Arial", Font.PLAIN, 10));
-            resultDescription.setFont(new Font("Arial", Font.PLAIN, 10));
-            resultDate.setFont(new Font("Arial", Font.PLAIN, 10));
-
-            resultTitle.setLocation(x_length + 200, y_height);
-            resultPrice.setLocation(x_length + 1000, y_height);
-            resultImage.setLocation(x_length, y_height);
-            resultDate.setLocation(x_length + 200, y_height + 25);
-            resultDescription.setLocation(x_length + 200, y_height + 50);
-
+    private JLabel titleLabel;
+    private final SpringLayout LAYOUT;
+    private SearchBarPanel SearchBar;
+    private JTextField jtSearch = new JTextField(20);
+    private final JButton SEARCH = new JButton("Search");
+    private DatabaseController controller = new DatabaseController();
 
-            resultTitle.setSize(300, 30);
-            resultPrice.setSize(100, 30);
-            resultImage.setSize(100, 100);
-            resultDescription.setSize(300, 100);
-            resultDate.setSize(200, 30);
-
-            y_height += 200;
+    public ListingListPage() throws IOException {
+        super("Scamazon.ca");
+        this.LAYOUT = new SpringLayout();
 
+        ArrayList<Listing> allListings = controller.getAllListings();
 
+        //displays all listings from database before User has searched anything
+        setUpPanel(allListings);
+    }
 
-            c.add(resultTitle);
-            c.add(resultDate);
-            c.add(resultDescription);
-            c.add(resultImage);
-            c.add(resultPrice);
+    private void setUpPanel(List<Listing> listings) {
+        this.setPreferredSize(new Dimension(1280, 720));
+        this.setLayout(LAYOUT);
+
+        //SearchBar with TextField and SEARCH Button
+        SEARCH.addActionListener(this);
+        SearchBar = new SearchBarPanel(jtSearch, SEARCH);
```

Comment:
So this is the search bar? On top of the listings right?

Reply:
we will have to test ui in class today